.build/build-sonar.xml

```diff
@@ -0,0 +1,158 @@
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project basedir="." name="apache-cassandra-sonar-tasks" xmlns:if="ant:if" xmlns:unless="ant:unless">
+    <property name="sonar-scanner-cli.version" value="5.0.1.3006-linux"/>
+    <property name="sonar.workdir" value="${build.dir}/sonar"/>
+    <property name="sonar.download.dir" value="${user.home}/.cassandra-cache/tools"/>
+    <property name="sonar-scanner.home" value="${sonar.download.dir}/sonar-scanner-${sonar-scanner-cli.version}"/>
+    <property name="sonar.projectKey" value="cassandra"/>
+    <property name="sonar.mainBranch" value="trunk"/>
+
+    <target name="sonar-init">
+        <mkdir dir="${sonar.download.dir}"/>
+        <mkdir dir="${sonar-scanner.home}"/>
+        <mkdir dir="${sonar.workdir}"/>
+        <mkdir dir="${sonar.workdir}/scanner"/>
+        <mkdir dir="${sonar.workdir}/data"/>
+        <mkdir dir="${sonar.workdir}/logs"/>
+        <mkdir dir="${sonar.workdir}/extensions"/>
+    </target>
+
+    <target name="init-sonar-scanner-cli" depends="sonar-init" unless="sonar-scanner.downloaded">
+        <local name="sonar-scanner.downloaded"/>
+        <available file="${sonar-scanner.home}/bin/sonar-scanner" property="sonar-scanner.downloaded"/>
+
+        <sequential unless:set="sonar-scanner.downloaded">
+            <echo>Downloading Sonar-Scanner CLI...</echo>
+            <retry retrycount="3" retrydelay="5000" >
+                <get src="https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-${sonar-scanner-cli.version}.zip"
+                     dest="${sonar.download.dir}/sonar-scanner-cli-${sonar-scanner-cli.version}.zip"/>
+            </retry>
+            <unzip src="${sonar.download.dir}/sonar-scanner-cli-${sonar-scanner-cli.version}.zip" dest="${sonar.download.dir}" overwrite="true"/>
+            <setpermissions mode="755">
+                <!-- unzip task in Ant does not preserve file permissions, so we need to make those files executable explicitly -->
+                <file file="${sonar-scanner.home}/bin/sonar-scanner" />
+                <file file="${sonar-scanner.home}/jre/bin/java" />
+                <file file="${sonar-scanner.home}/jre/lib/jspawnhelper" />
+            </setpermissions>
+        </sequential>
+    </target>
+
+    <macrodef name="check-server-status">
+        <attribute name="url"/>
+        <sequential>
+            <exec executable="bash" failifexecutionfails="true" failonerror="true" logerror="false">
+                <arg value="-c"/>
+                <arg value="curl -s &quot;@{url}/api/system/status&quot; | jq -r &quot;.status&quot; | grep UP"/>
```

Comment:
I am not saying that depending on jq is wrong, I am just saying that people will probably have to install this on their machines beforehand. If we dont replace it with something else, maybe some "sanity check" target to check if all required tooling is available before advancing this target as a whole would be nice so it fails as soon as possible.

Reply:
`jq` will be probably needed for OWASP as well... we can maybe use this: https://github.com/eiiches/jackson-jq

Reply:
+1 yeah probably

Reply:
thinking about that more... I'm rather reluctant to that. It is not very uncommon thing to have `jq` installed when you are a developer. I'm ok with failing clearly if this prerequisite is not satisfied.

Reply:
OK do it like that then.